yt/data_objects/static_output.py

```diff
@@ -1934,6 +1934,17 @@ def __init__(
             default_species_fields=default_species_fields,
         )
 
+    def _disable_periodicity_for_bbox(self):
+        # This is called when a bounding box is explicitly set by the
+        # user--currently in these circumstances periodicity behaves
+        # very oddly (see https://github.com/yt-project/yt/issues/2639),
+        # so to prevent buggy behavior we simply disable it.
+        mylog.warning(
+            "A bounding box was explicitly specified, so we "
+            "are disabling periodicity."
+        )
+        self._periodicity += (False,) * 3
```

Comment:
I believe this should solve the failures
```suggestion
        self._periodicity = (False,) * 3
```

Reply:
Whoops, thanks